internal/engine/page/page.go

```diff
@@ -260,25 +260,30 @@ func (p *Page) storeRawCell(rawCell []byte) error {
 // This assumes, that the cell data has already been inserted into the page at the given
 // slot.
 func (p *Page) storeCellSlot(slot Slot) {
-	offsets := append(p.OccupiedSlots(), slot)
-	sort.Slice(offsets, func(i, j int) bool {
-		var leftKey, rightKey []byte
-		switch c := p.CellAt(offsets[i]).(type) {
-		case RecordCell:
-			leftKey = c.Key
-		case PointerCell:
-			leftKey = c.Key
-		}
-		switch c := p.CellAt(offsets[j]).(type) {
+	slots := p.OccupiedSlots()
+	var pivotKey []byte
+	switch c := p.CellAt(slot).(type) {
+	case RecordCell:
+		pivotKey = c.Key
+	case PointerCell:
+		pivotKey = c.Key
+	}
+
+	insertIndex := sort.Search(len(slots), func(i int) bool {
```

Comment:
Searching for an insert index opposed to sorting all slots and rewriting all of them improved insert performance by about 45%.
For 100_000 inserts, the time dropped from 20sec to 11sec.
Maybe this can be improved even more in the future.

Reply:
We don't search for an existing slot. The method `sort.Search` returns an index, at which an element must be inserted (for the slice to remain sorted).
Inserting the element correctly and then storing all new slots on disk is faster and better than obtaining all slots, adding the new one to the end, sorting the whole slice (which is what we did before) and then writing them back to disk.